internal/distro/rhel8/azure.go

```diff
@@ -695,6 +683,65 @@ var defaultAzureRhuiImageConfig = &distro.ImageConfig{
 			},
 		},
 	},
+	EnabledServices: append(defaultAzureImageConfig.EnabledServices, "firewalld"),
+}
+
+const wildflyPath = "/opt/rh/eap7/root/usr/share/wildfly"
+
+var defaultAzureEapImageConfig = &distro.ImageConfig{
+	GPGKeyFiles: []string{
+		"/etc/pki/rpm-gpg/RPM-GPG-KEY-microsoft-azure-release",
+		"/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release",
+	},
+	RHSMConfig: map[distro.RHSMSubscriptionStatus]*osbuild.RHSMStageOptions{
+		distro.RHSMConfigNoSubscription: {
+			DnfPlugins: &osbuild.RHSMStageOptionsDnfPlugins{
+				SubscriptionManager: &osbuild.RHSMStageOptionsDnfPlugin{
+					Enabled: false,
+				},
+			},
+			SubMan: &osbuild.RHSMStageOptionsSubMan{
+				Rhsmcertd: &osbuild.SubManConfigRHSMCERTDSection{
+					AutoRegistration: common.ToPtr(true),
+				},
+				Rhsm: &osbuild.SubManConfigRHSMSection{
+					ManageRepos: common.ToPtr(false),
+				},
+			},
+		},
+		distro.RHSMConfigWithSubscription: {
+			SubMan: &osbuild.RHSMStageOptionsSubMan{
+				Rhsmcertd: &osbuild.SubManConfigRHSMCERTDSection{
+					AutoRegistration: common.ToPtr(true),
+				},
+				// do not disable the redhat.repo management if the user
+				// explicitly request the system to be subscribed
+			},
+		},
+	},
```

Comment:
I realized that I made things worse.

I think that these should be omitted from the EAP-specific config, since these are RHUI-specific.

I suggest to:

- remove this block from the EAP config.
- remove the appending of `firewalld` service in the RHUI / BYOS configs.
- except for the EAP image, replace all instances when inheriting from `defaultAzureImageConfig` to inherit from `vhdImageConfig()`. This will make sure that the `firewalld` service is enabled.
- for the EAP image, use `defaultAzureEapImageConfig.InheritFrom(defaultAzureRhuiImageConfig.InheritFrom(defaultAzureImageConfig))`. This will keep the EAP config simple and also remove the need to duplicate the RHUI-specific config in two places.

Reply:
Yeah you're right.  I made a mess earlier.  Cleaned up now.